tensorflow_datasets/scripts/generate_visualization.py

```diff
@@ -0,0 +1,62 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Script to generate datasets figures and save them
+  in docs/catalog/images/ directory.
+
+Args:
+  data_name: name of the dataset, for which figures to be generated.
+"""
+
+import os
+from absl import app
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+FIG_DIR = os.path.join('..', 'docs', 'catalog', 'images')
+FULL_PATH = tfds.core.get_tfds_path(FIG_DIR)
+
+def generate_vizz(data_name):
+    """Save the generated figures for the dataset
+    Args:
+      data_name: name of the dataset
+    """
+    try:
+        print("Generating examples...")
+        builder = tfds.builder(data_name)
```

Comment:
You should create one image per-config.

Reply:
In `document_dataset_examples.py` file I create a list of dataset having config or without config. From their I call this `generate_visualization.py` which will take dataset with configs and save image in dest_sir. Is it okay?

Reply:
The document_dataset_examples.py is nice for you to test, but won't be merged to TFDS.

We want a standalone script which generates the dataset config for all the datasets. similarly to 
 https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/scripts/document_datasets.py

So you should create a new function:
```
def generate_visualization(datasets=None):
```
which if None loads all datasets and configs from all datasets.

And then you can pass those names to your `generate_vizz` function (that you could rename `generate_single_visualization(ds_name)`)

Reply:
I am applying `ThreadPoolExecutor` to generate builders like this:

```py
with futures.ThreadPoolExecutor(max_workers=WORKER_COUNT_DATASETS) as tpool:
            builder_examples = tpool.map(generate_single_visualization, builders)
```
And then in `generate_single_visualization` I am making list of all datasets with or without config like
```py
def generate_single_visualization(builder):
    """Save the generated figures for the dataset
    Args:
      data_name: name of the dataset
    """
    dataset_config_list = []
    if builder.builder_configs:
        for config in builder.BUILDER_CONFIGS:
            dataset_config_list.append(os.path.join(builder.name, config.name))
    else:
        dataset_config_list.append(builder.name)
```
So just want to ask is this is okay or I have to generate configs also using `ThreadPoolExecutor` ?

Reply:
I think the builder config should be instantiated inside the thread pool, as loading the builder may add some overhead that we want to paralellize.
Also I think we want to parallelize the configs from the same builder.

So the best would be to have something like:
```
tpool.map(generate_single_visualization, ['dataset0/config0', 'dataset0/config1', 'dataset1',...])
```

To generate the list of config names, the best would be to add a new function in `registered.py` like:

```python
def list_config_names(predicate_fn: Callable[Type[DatasetBuilder], bool] = None) -> List[str]:
  """Lists all registered `dataset/config` and `dataset` names."""
  # Strip the version from the full names
  config_names = {n.rsplit('/', maxsplit=1)[0] for n in list_full_names(predicate_fn)}
  return sorted(config_names)
```

Reply:
Thanks for help, I will make changes, So after this `list_config_names` I have to pass this functions return list instead `['dataset0/config0', 'dataset0/config1', 'dataset1',...]`